packages/components/src/select-control/types.ts

```diff
@@ -58,7 +58,7 @@ export interface SelectControlProps
 		 */
 		disabled?: boolean;
 	}[];
-	value?: string | string[];
+	value?: string | string[] | number;
```

Comment:
The `<QueryControls>` component passes to the `<SelectControl>` a `value` [prop](https://github.com/WordPress/gutenberg/blob/921135f9ec7f1e54030672eb7d11ca28034bf15b/packages/components/src/select-control/index.tsx#L122) of type `number`.

<img width="1031" alt="Screen Shot 2022-12-21 at 4 59 48 PM" src="https://user-images.githubusercontent.com/4063887/209019824-47b47589-980f-4889-9db5-c0085f625695.png">

And the React type [SelectHTMLAttributes ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/04d5ce209125122ab55ad7502c88a5ef05c1b9a8/types/react/index.d.ts#L2412) allows a `value` of `number`:

```ts
value?: string | ReadonlyArray<string> | number | undefined;
```

Reply:
Otherwise, I could fix this typing issue in [packages/components/src/query-controls](https://github.com/WordPress/gutenberg/tree/921135f9ec7f1e54030672eb7d11ca28034bf15b/packages/components/src/query-controls)

But that'll be more complex.

Reply:
Sounds good, how does this look? https://github.com/WordPress/gutenberg/pull/46721/commits/2442dd543021b311896131673b755043ff9c0456#diff-839b843623758f86e150a4405d4e513833b7e9b5413439ed082f44d970cd11f3R21

Reply:
Much better! Although we still need to guard against `undefined`, because `String( undefined )` would print `'undefined'`.

<details>

<summary>Maybe something like this?</summary>

```diff
diff --git a/packages/components/src/query-controls/author-select.tsx b/packages/components/src/query-controls/author-select.tsx
index 03e6b07d29..37163aba4d 100644
--- a/packages/components/src/query-controls/author-select.tsx
+++ b/packages/components/src/query-controls/author-select.tsx
@@ -18,7 +18,11 @@ export default function AuthorSelect( {
 		<TreeSelect
 			{ ...{ label, noOptionLabel, onChange } }
 			tree={ termsTree }
-			selectedId={ String( selectedAuthorId ) }
+			selectedId={
+				selectedAuthorId !== undefined
+					? String( selectedAuthorId )
+					: undefined
+			}
 		/>
 	);
 }
diff --git a/packages/components/src/query-controls/category-select.tsx b/packages/components/src/query-controls/category-select.tsx
index d96763b7b4..c082e3d977 100644
--- a/packages/components/src/query-controls/category-select.tsx
+++ b/packages/components/src/query-controls/category-select.tsx
@@ -26,7 +26,11 @@ export default function CategorySelect( {
 		<TreeSelect
 			{ ...{ label, noOptionLabel, onChange } }
 			tree={ termsTree }
-			selectedId={ String( selectedCategoryId ) }
+			selectedId={
+				selectedCategoryId !== undefined
+					? String( selectedCategoryId )
+					: undefined
+			}
 			{ ...props }
 		/>
 	);
```

</details>

Reply:
Good point.

https://github.com/WordPress/gutenberg/pull/46721/commits/284051750cde774c2db36f60cc6b2e7bcd5ec953 commits your suggestion verbatim.